cmd/admin-config-export.go

```diff
@@ -89,7 +90,7 @@ func mainAdminConfigExport(ctx *cli.Context) error {
 
 	// Print
 	printMsg(configExportMessage{
-		Value: buf,
+		Value: fmt.Sprintf("%s", buf),
```

Comment:
The original reason it's not converted to a string is to ensure that the representation of the `export` command doesn't need to be jsonified and base64 representation keeps the entire file in-tact including its `\n` for further consumption

```
~ echo "cmVnaW9uIG5hbWU9dXMtd2VzdC0xIApzdG9yYWdlX2NsYXNzIHN0YW5kYXJkPSBycnM9RUM6MiAKIyBjYWNoZSBkcml2ZXM9IGV4Y2x1ZGU9IGV4cGlyeT05MCBxdW90YT04MCAKIyBjb21wcmVzc2lvbiBlbmFibGU9b2ZmIGV4dGVuc2lvbnM9LnR4dCwubG9nLC5jc3YsLmpzb24sLnRhciwueG1sLC5iaW4gbWltZV90eXBlcz10ZXh0LyosYXBwbGljYXRpb24vanNvbixhcHBsaWNhdGlvbi94bWwgCiMgZXRjZCBlbmRwb2ludHM9IHBhdGhfcHJlZml4PSBjb3JlZG5zX3BhdGg9L3NreWRucyBjbGllbnRfY2VydD0gY2xpZW50X2NlcnRfa2V5PSAKIyBpZGVudGl0eV9vcGVuaWQgY29uZmlnX3VybD0gY2xpZW50X2lkPSBjbGFpbV9uYW1lPXBvbGljeSBjbGFpbV9wcmVmaXg9IGp3a3NfdXJsPSAKIyBwb2xpY3lfb3BhIHVybD0gYXV0aF90b2tlbj0gCiMga21zX3ZhdWx0IGVuZHBvaW50PSBrZXlfbmFtZT0gYXV0aF90eXBlPWFwcHJvbGUgYXV0aF9hcHByb2xlX2lkPSBhdXRoX2FwcHJvbGVfc2VjcmV0PSBjYXBhdGg9IGtleV92ZXJzaW9uPSBuYW1lc3BhY2U9IAojIGttc19rZXMgZW5kcG9pbnQ9IGtleV9uYW1lPSBjZXJ0X2ZpbGU9IGtleV9maWxlPSBjYXBhdGg9IAojIGxvZ2dlcl93ZWJob29rIGVuYWJsZT1vZmYgZW5kcG9pbnQ9IGF1dGhfdG9rZW49IAojIGF1ZGl0X3dlYmhvb2sgZW5hYmxlPW9mZiBlbmRwb2ludD0gYXV0aF90b2tlbj0gCiMgbm90aWZ5X2FtcXAgZW5hYmxlPW9mZiB1cmw9IGV4Y2hhbmdlPSBleGNoYW5nZV90eXBlPSByb3V0aW5nX2tleT0gbWFuZGF0b3J5PW9mZiBkdXJhYmxlPW9mZiBub193YWl0PW9mZiBpbnRlcm5hbD1vZmYgYXV0b19kZWxldGVkPW9mZiBkZWxpdmVyeV9tb2RlPTAgcXVldWVfbGltaXQ9MCBxdWV1ZV9kaXI9IAojIG5vdGlmeV9rYWZrYSBlbmFibGU9b2ZmIHRvcGljPSBicm9rZXJzPSBzYXNsX3VzZXJuYW1lPSBzYXNsX3Bhc3N3b3JkPSBjbGllbnRfdGxzX2NlcnQ9IGNsaWVudF90bHNfa2V5PSB0bHNfY2xpZW50X2F1dGg9MCBzYXNsPW9mZiB0bHM9b2ZmIHRsc19za2lwX3ZlcmlmeT1vZmYgcXVldWVfbGltaXQ9MCBxdWV1ZV9kaXI9IAojIG5vdGlmeV9tcXR0IGVuYWJsZT1vZmYgYnJva2VyPSB0b3BpYz0gcGFzc3dvcmQ9IHVzZXJuYW1lPSBxb3M9MCBrZWVwX2FsaXZlX2ludGVydmFsPTBzIHJlY29ubmVjdF9pbnRlcnZhbD0wcyBxdWV1ZV9kaXI9IHF1ZXVlX2xpbWl0PTAgCiMgbm90aWZ5X25hdHMgZW5hYmxlPW9mZiBhZGRyZXNzPSBzdWJqZWN0PSB1c2VybmFtZT0gcGFzc3dvcmQ9IHRva2VuPSB0bHM9b2ZmIHRsc19za2lwX3ZlcmlmeT1vZmYgY2VydF9hdXRob3JpdHk9IGNsaWVudF9jZXJ0PSBjbGllbnRfa2V5PSBwaW5nX2ludGVydmFsPTAgc3RyZWFtaW5nPW9mZiBzdHJlYW1pbmdfYXN5bmM9b2ZmIHN0cmVhbWluZ19tYXhfcHViX2Fja3NfaW5fZmxpZ2h0PTAgc3RyZWFtaW5nX2NsdXN0ZXJfaWQ9IHF1ZXVlX2Rpcj0gcXVldWVfbGltaXQ9MCAKIyBub3RpZnlfbnNxIGVuYWJsZT1vZmYgbnNxZF9hZGRyZXNzPSB0b3BpYz0gdGxzPW9mZiB0bHNfc2tpcF92ZXJpZnk9b2ZmIHF1ZXVlX2Rpcj0gcXVldWVfbGltaXQ9MCAKIyBub3RpZnlfbXlzcWwgZW5hYmxlPW9mZiBmb3JtYXQ9bmFtZXNwYWNlIGhvc3Q9IHBvcnQ9IHVzZXJuYW1lPSBwYXNzd29yZD0gZGF0YWJhc2U9IGRzbl9zdHJpbmc9IHRhYmxlPSBxdWV1ZV9kaXI9IHF1ZXVlX2xpbWl0PTAgCiMgbm90aWZ5X3Bvc3RncmVzIGVuYWJsZT1vZmYgZm9ybWF0PW5hbWVzcGFjZSBjb25uZWN0aW9uX3N0cmluZz0gdGFibGU9IGhvc3Q9IHBvcnQ9IHVzZXJuYW1lPSBwYXNzd29yZD0gZGF0YWJhc2U9IHF1ZXVlX2Rpcj0gcXVldWVfbGltaXQ9MCAKIyBub3RpZnlfcG9zdGdyZXM6aG9saSBlbmFibGU9b2ZmIGZvcm1hdD1uYW1lc3BhY2UgY29ubmVjdGlvbl9zdHJpbmc9IHRhYmxlPSBob3N0PSBwb3J0PSB1c2VybmFtZT0gcGFzc3dvcmQ9IGRhdGFiYXNlPSBxdWV1ZV9kaXI9IHF1ZXVlX2xpbWl0PTAgCiMgbm90aWZ5X3Bvc3RncmVzOnRlc3QgZW5hYmxlPW9mZiBmb3JtYXQ9bmFtZXNwYWNlIGNvbm5lY3Rpb25fc3RyaW5nPSB0YWJsZT0gaG9zdD0gcG9ydD0gdXNlcm5hbWU9IHBhc3N3b3JkPSBkYXRhYmFzZT0gcXVldWVfZGlyPSBxdWV1ZV9saW1pdD0wIAojIG5vdGlmeV9lbGFzdGljc2VhcmNoIGVuYWJsZT1vZmYgdXJsPSBmb3JtYXQ9bmFtZXNwYWNlIGluZGV4PSBxdWV1ZV9kaXI9IHF1ZXVlX2xpbWl0PTAgCiMgbm90aWZ5X3JlZGlzIGVuYWJsZT1vZmYgZm9ybWF0PW5hbWVzcGFjZSBhZGRyZXNzPSBrZXk9IHBhc3N3b3JkPSBxdWV1ZV9kaXI9IHF1ZXVlX2xpbWl0PTAg" | base64 --decode
```

Making it string makes it tricky to now consume it properly, requires additionally trickery.  I would say why not avoid this change in this PR?

We can add another version of `madmin.ParseSubSysTarget` which simply returns the correct representation of data structure which can be stringified etc and also when jsonified converts into proper JSON.

Reply:
yeah, that works